aips/objects.md

```diff
@@ -13,22 +13,21 @@ requires: AIP_number_of(Resource Groups)
 
 ## Summary
 
-This AIP proposes Move objects for global access to composite set of resources stored at a single address on-chain. By leveraging the account model, objects can directly emit events that can lead to richer understanding of on-chain actions. Finally, objects offer a rich capability model that allows for fine-grained resource control and ownership objects in a single store.
+This AIP proposes Move objects for global access to heterogeneous set of resources stored at a single address on-chain. By leveraging the aspects of the account model, objects can directly emit events that can lead to richer understanding of on-chain actions. Finally, objects offer a rich capability model that allows for fine-grained resource control and ownership objects in a single store.
 
 ## Motivation
 
-The object model allows Move on Aptos to represent a complex type as a set of resources stored within a single address. In an object model, an NFT or token can place common token data within a `Token` resource, object data within an `Object` resource, and then specialize into additional resources as necessary, for example, a `Player` object could define a player within a game. 
+The object model allows Move on Aptos to represent a complex type as a set of resources stored within a single address. In an object model, an NFT or token can place common token data within a `Token` resource, object data within an `Object` resource, and then specialize into additional resources as necessary, for example, a `Player` object could define a player within a game. The `Object` itself stores both the `address` of the current owner, the appropriate data for creating event streams, and a value to denote if the object has been deleted.
 
 The object model improves type safety and data access through the use of a capability framework implemented by different accessors or references that define and distribute out various capabilities that allow for rich operations. These include:
 
-- `CreatorRef` that allows the creation of all other capabilities, allows adding resources to the object, and can only be accessed at the time the object is created
-- `OwnerRef` defines ownership of an object. Objects may have one or more `OwnerRef` and an `OwnerRef` can be aggregated, thus allow for fungability of objects.
-- `MintRef` allows for the creation of `OwnerRef`.
+- `CreatorRef` that allows the creation of all other capabilities, allows adding resources to the object, and can only be accessed at the time the object is created.
```

Comment:
the other refs can be stored, and in sum do the same thing basically that `CreatorRef` do, except without being able to get the signer. Can we please consider adding `store` to `CreatorRef` as well? I understand there is a design pattern you are thinking of here, but to give creators options, having `store` might be the most flexible

Reply:
Considering that there's really only two ref types that creator can leverage (Delete and Transfer), I think this isn't worth spending too much time on at this juncture.

A major reason to not allow creator ref is that it provides the same backdoor functionality of allowing a creator of an nft to specify that an nft can be burned after mint.

I think it is important to make it clear intentionality when creating something. If I claim something is immutable and forever the data and the API associated with that object should be easily verifiable as such.